src/components/Transaction/Transaction.tsx

```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { ReactNode, ComponentPropsWithoutRef, cloneElement } from 'react';
+
+export interface TransactionProps extends ComponentPropsWithoutRef<'div'> {
+  preIcon: JSX.Element;
+  title: string;
+  address: string;
+  posIcon: JSX.Element;
+  amount: string;
+  timestamp: string;
+  customClass?: string;
+  variant?: 'secondary' | undefined;
+  children?: ReactNode;
+}
+
+interface Classes {
+  [key: string]: string | object;
+}
+
+export function Transaction(props: TransactionProps) {
+  let {
+    preIcon,
+    title,
+    address,
+    posIcon,
+    amount,
+    timestamp,
+    variant = 'primary',
+    children,
+    customClass,
+    ...rest
+  } = props;
+
+  const CLASSES: Classes = {
+    root: 'flex justify-center items-center',
+    primary: {
+      default: 'default-primary',
+      preIcon: 'bg-primary text-f-primary rounded-full p-1 h-8 w-8 mr-2',
+    },
+    secondary: {
+      default: 'default-secondary',
+      preIcon: 'bg-neutral text-f-secondary rounded-full p-1 h-8 w-8 mr-2',
+    },
+  };
+
+  const VARIANT = CLASSES[variant] as Classes;
+
+  const clonedPreIcon = preIcon
+    ? cloneElement(preIcon, {
+        className: VARIANT.preIcon,
+      })
+    : null;
+
+  const date = new Date(timestamp);
+  const options: Intl.DateTimeFormatOptions = {
+    weekday: 'long',
+    hour: '2-digit',
+    minute: '2-digit',
+    hour12: true,
+  };
+  const formattedDate = date
+    .toLocaleString('en-US', options)
+    .replace(/(\w+)(\s+)/, '$1, ');
```

Comment:
I think we can deal with date here as simple string and let's the component where we place the `<Trasaction>` deal with the format of it.

Reply:
done